lib/c_glib/src/thrift/c_glib/protocol/thrift_compact_protocol.c

```diff
@@ -120,7 +120,11 @@ thrift_bitwise_cast_gdouble (const guint64 v)
 static guint64
 i64_to_zigzag (const gint64 l)
 {
-  return (l << 1) ^ (l >> 63);
```

Comment:
Well, you can only put those suffixes like "ULL" on literals, and the only literals here are the right-hand sides of the shifts, and the right-hand sides are not the problem. The problem is that `p << q` has undefined behavior when p is negative.